service/s3/s3manager/upload.go

```diff
@@ -614,58 +622,21 @@ func (u *multiuploader) upload(firstBuf io.ReadSeeker) (*UploadOutput, error) {
 	}
 	u.uploadID = *resp.UploadId
 
-	// Create the workers
```

Comment:
I think we can keep the old code and simply use [`sync.Pool`](https://golang.org/pkg/sync/#Pool). What do you think? This will get rid of the mutexes and critical sections.

Reply:
I like it. This would work and probably end up being much less code change. I'll rework this and update asap.